src/util.hpp

```diff
@@ -52,6 +52,26 @@ int less(double x, double y);
  */
 int greater(double x, double y);
 
+/**
+ * Return 1 if the number if positive and 0 otherwise.
```

Comment:
Is 0 considered positive? Also when the value is negative, is the returned value 0.0 or -1.0?

Reply:
According to what I understand, 0.0 is considered as negative sign for Jacobi rotations.
Ref: http://www2.isec.pt/~armenioc/privado/PapersLivrosSites/Matlab/Cambridge%20University%20Press%20-%20Numerical%20Methods%20in%20Engineering%20with%20MATLAB/Cambridge.University.Press.Numerical.Methods.in.Engineering.with.MATLAB.Aug.2005.eBook-DDU.pdf

Pages : 331 and 332


I'm slightly modifying that function to return an int value instead of double.

Reply:
- Ok! I wil check if also other sources consider sign(x) = -1.0 when x<=0.
- Regarding returning an int, if you are gonna use this 1 or -1 in a multiplication with double, why return an int instead of a double.
- Also the comment should be updated as right now it mentions return values of 1 and 0.

Reply:
Well, most of the implementations of sign() function are designed to return an int, and I followed it assuming that there will be no NaN kind of inputs (Ref: https://stackoverflow.com/a/35491710) and int takes less space and has faster processing speeds than floating points (a kind of optimization if you wanna consider it in that way ;-) ). But it's debatable though.

I've updated the comment too....

Reply:
I agree that int takes may be better in some cases but I don't think it is better in our case. Imagine you do `c = sign(a) * b` and `b` and `c` are doubles. The compiler will generate an extra instruction to convert the int returned by `sign(a)` to double.

Reply:
That's true though. We could as well have multiple versions of such utility functions returning different data types, but that becomes too arcane at least for now. Will revert it back to return double.

Reply:
A simple template function should suffice